ProjFS.Linux/PrjFSLib.Linux.Managed/VirtualizationInstance.cs

```diff
@@ -235,6 +238,50 @@ public virtual void StopVirtualizationInstance()
             throw new NotImplementedException();
         }
 
+        private static Result RemoveFileOrDirectory(
+            string fullPath,
+            bool isDirectory)
+        {
+            try
+            {
+                if (isDirectory)
+                {
+                    Directory.Delete(fullPath);
+                }
+                else
+                {
+                    File.Delete(fullPath);
+                }
+            }
+            catch (IOException ex) when (ex is DirectoryNotFoundException)
+            {
+                return Result.EPathNotFound;
+            }
+            catch (IOException ex) when (ex is FileNotFoundException)
+            {
+                return Result.EFileNotFound;
+            }
+            catch (IOException ex)
```

Comment:
For consistency with above you could consider `catch (IOException ex) when (ex.HResult == Errno.Constants.ENOTEMPTY)`.

Reply:
Good idea!  Done!